Sources/CoreUITestKit/XCUIElement+.swift

```diff
@@ -118,62 +122,93 @@ extension XCUIElement {
                 object: self
             )
             let result = XCTWaiter.wait(for: [expectation], timeout: timeout)
-            let checkHittable = !self.dsl_isHittable || result == .completed
-            return checkHittable
+            return self.dsl_isHittable || (result == .completed)
         }
     }
-    
-    public func dsl_typeText(_ text: String, file: StaticString = #file, line: UInt = #line) {
+
+    public func dsl_typeText(_ text: String, needWait: Bool = false, file: StaticString = #file, line: UInt = #line) {
         let resourceName = self.description
         XCTContext.runActivity(named: "Ввод текста \(text) в \(String(describing: resourceName))") { _ in
-            guard dsl_isHittable else {
-                XCTFail ("Элемент \(String(describing: resourceName)) не готов для нажатия",
-                file: file,
-                line: line
-                )
-                return
+            if needWait {
+                guard dsl_waitForHittable(timeout: 5) else {
+                    XCTFail(
+                        "Элемент \(String(describing: resourceName)) не готов для нажатия",
+                        file: file,
+                        line: line
+                    )
+                    return
+                }
+                XCTContext.runActivity(named: "Ввод текста \(text)") { _ in
+                    typeText(text)
+                }
+            } else {
+                XCTContext.runActivity(named: "Ввод текста \(text)") { _ in
+                    typeText(text)
+                }
```

Comment:
кажется можно вот так написать? (и в похожих кейсах ниже)
```suggestion
            if needWait && !dsl_waitForHittable(timeout: 5) {
                XCTFail(
                    "Элемент \(String(describing: resourceName)) не готов для нажатия",
                    file: file,
                    line: line
                )
            } else {
                XCTContext.runActivity(named: "Ввод текста \(text)") { _ in
                    typeText(text)
                }
```